lib/connectors/highlevel_client.go

```diff
@@ -0,0 +1,341 @@
+	return c.client.GetConnectorConfig(req)
+}
+
+//GetConnectorStatus return current status of connector
+func (c *highLevelClient) GetConnectorStatus(req ConnectorRequest) (GetConnectorStatusResponse, error) {
+	return c.client.GetConnectorStatus(req)
+}
+
+//RestartConnector restart connector
+func (c *highLevelClient) RestartConnector(req ConnectorRequest) (EmptyResponse, error) {
+	return c.client.RestartConnector(req)
+}
+
+//PauseConnector pause a running connector
+//asynchronous operation
+func (c *highLevelClient) PauseConnector(req ConnectorRequest, sync bool) (EmptyResponse, error) {
+	result, err := c.client.PauseConnector(req)
+	if err != nil {
+		return result, err
+	}
+
+	if sync {
+		if !tryUntil(
+			func() bool {
+				resp, err := c.GetConnectorStatus(req)
+				return err == nil && resp.Code == 200 && resp.ConnectorStatus["state"] == "PAUSED"
+			},
+			2*time.Minute,
+		) {
+			return result, errors.New("timeout on pausing connector sync")
+		}
+	}
+	return result, nil
+}
+
+//ResumeConnector resume a paused connector
+//asynchronous operation
+func (c *highLevelClient) ResumeConnector(req ConnectorRequest, sync bool) (EmptyResponse, error) {
+	result, err := c.client.ResumeConnector(req)
+	if err != nil {
+		return result, err
+	}
+
+	if sync {
+		if !tryUntil(
+			func() bool {
+				resp, err := c.GetConnectorStatus(req)
+				return err == nil && resp.Code == 200 && resp.ConnectorStatus["state"] == "RUNNING"
+			},
+			2*time.Minute,
+		) {
+			return result, errors.New("timeout on resuming connector sync")
+		}
+	}
+	return result, nil
+}
+
+//IsUpToDate checks if the given configuration is different from the deployed one.
+//Returns true if they are the same
+func (c *highLevelClient) IsUpToDate(connector string, config map[string]interface{}) (bool, error) {
+	// copy the map to safely interact with it
+	// we are going to need to add connector name to be able to exact match
+	copyConfig := make(map[string]interface{}, len(config))
+	for key, value := range config {
+		copyConfig[key] = value
+	}
+
+	copyConfig["name"] = connector
+
+	configResp, err := c.GetConnectorConfig(ConnectorRequest{Name: connector})
+	if err != nil {
+		return false, err
+	}
+	if configResp.Code == 404 {
+		return false, nil
+	}
+	if configResp.Code >= 400 {
+		return false, errors.New(fmt.Sprintf("status code: %d", configResp.Code))
+	}
+
+	if len(configResp.Config) != len(copyConfig) {
+		return false, nil
+	}
+	for key, value := range configResp.Config {
+		if convertConfigValueToString(copyConfig[key]) != convertConfigValueToString(value) {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// Because trying to compare the same field on 2 different config may return false negative if one is encoded as a string and not the other
+func convertConfigValueToString(value interface{}) string {
+	switch v := value.(type) {
+	case string:
+		return v
+	case int:
```

Comment:
You might miss int32, int64, etc... (don't know if it can happen).
Stupid question: why not use `fmt.Sprintf("%v", value)`?

Reply:
this is code is a bit old ( I just moved it in this PR)
But you may be right, `fmt.Sprintf` seem safer.

Will see if I an replace